gammapy/modeling/parameter.py

```diff
@@ -499,15 +515,50 @@ def autoscale(self):
 
         """
         if self.scale_method == "scale10":
-            value = self.value
             if value != 0:
                 exponent = np.floor(np.log10(np.abs(value)))
-                scale = np.power(10.0, exponent)
-                self.factor = value / scale
-                self.scale = scale
+                self._scale = np.power(10.0, exponent)
 
         elif self.scale_method == "factor1":
-            self.factor, self.scale = 1, self.value
+            self._scale = value
+
+    def transform(self, value, update_scale=False):
+        """Tranform from value to factor (used by the optimizer).
+
+        Parameters
+        ----------
+        value : float
+            Parameter value
+        update_scale : bool, optional
+            Update the scaling (used by the autoscale). Default is False.
+        """
+        interp_scale = interpolation_scale(self.scale_transform)
+        factor = interp_scale(value)
```

Comment:
I understand this is not the same `factor` as before, and I must say this is confusing when trying to understand the code.

Reply:
renamed transformed